waiter/src/waiter/state.clj

```diff
@@ -1303,3 +1308,114 @@
     (t/millis router-syncer-interval-ms)
     #(retrieve-peer-routers discovery router-chan)
     :delay-ms router-syncer-delay-ms))
+
+;; Support for tracking instance scheduling/startup time stats
+
+(defrecord InstanceTracker [service-id known-instance-ids scheduling-instance-timer-contexts starting-instance-trackers])
+
+(defn- new-tracker-val
+  [service-id service-id->service-description-fn]
+  (let [{:strs [min-instances]} (service-id->service-description-fn service-id)
+        ctxs (repeatedly
+               min-instances
+               (fn [] {:schedule [(timers/start (metrics/waiter-timer "service-init-overhead" "schedule-time"))
+                                  (timers/start (metrics/service-timer service-id "service-init-overhead" "schedule-time"))]
```

Comment:
Should we use a map here for the two timer contexts?

Reply:
I think the only attribute of the timers that we care about is when they should be stopped (once scheduled, or once healthy). After creation, we don't really care if the timer is service-level or waiter-level. Unless we care about accessing the timers individually, I don't think a map makes sense. I don't even care about the order here—but a vector still felt like a better fit than a set.